js/ui/status/keyboard.js

```diff
@@ -453,6 +493,27 @@ const InputSourceManager = new Lang.Class({
             this._mruSourcesBackup = null;
         }
 
+        // Initialize from settings when we have no MRU sources list
+        if (this._mruSources.length == 0) {
+            let mruSettings = this._settings.mruSources;
+            for (let i = 0; i < mruSettings.length; i++) {
+                let mruSettingSource = mruSettings[i];
+                let mruSource = null;
+
+                for (let j = 0; j < sourcesList.length; j++) {
+                    let source = sourcesList[j];
+                    if (source.type == mruSettingSource.type &&
+                        source.id == mruSettingSource.id) {
+                        mruSource = source;
```

Comment:
That's true... But since this was pushed upstream like this already, I prefer merging it as-is to avoid carrying an additional delta.